src/include/platform/DeviceControlServer.h

```diff
@@ -91,13 +97,25 @@ class DeviceControlServer final
     CHIP_ERROR DisarmFailSafe();
     CHIP_ERROR CommissioningComplete();
     CHIP_ERROR SetRegulatoryConfig(uint8_t location, const CharSpan & countryCode, uint64_t breadcrumb);
-
     CHIP_ERROR ConnectNetworkForOperational(ByteSpan networkID);
 
-    inline FabricIndex GetFabricIndex() { return mFabric; }
-    inline void SetFabricIndex(FabricIndex fabricId) { mFabric = fabricId; }
+    /**
+     * @brief
+     *   Matches the fail-safe context’s Fabric Index (including null Fabric Index for a just established PASE session).
+     */
+    bool MatchFabricIndex(FabricIndex accessingFabric)
+    {
+        return (accessingFabric == mFailSafeContextl.mFabric) || (mFailSafeContextl.mFabric == kUndefinedFabricIndex);
```

Comment:
This check does not look right.  It should just be doing `accessingFabricIndex == mFailSafeContext.mFabricIndex`, period, I would think.